official/recommendation/ncf_dataset.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Prepare dataset for NCF.
+
+Load the training dataset and evaluation dataset from csv file into memory.
+Prepare input for model training and evaluation.
+"""
+import numpy as np
+import scipy.sparse as sp
+from six.moves import xrange  # pylint: disable=redefined-builtin
+
+
+def load_train_matrix(train_fname):
+  """Load the train matrix from a given file.
+
+  Args:
+    train_fname: The csv file name of training dataset.
+
+  Returns:
+    mat: The sparse matrix of training data.
+  """
+  def process_line(line):
+    tmp = line.split('\t')
+    # user, item, rating
+    return [int(tmp[0]), int(tmp[1]), float(tmp[2]) > 0]
+
+  lines = open(train_fname, 'r').readlines()[1:]
+  data = list(map(process_line, lines))
```

Comment:
I generally prefer list comprehension to map() because the behavior changes from 2 to 3.